pkg/components/util/delete.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"helm.sh/helm/v3/pkg/action"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
+)
+
+// DeleteComponent deletes a component
+func DeleteComponent(name string, c components.Component, kubeconfig string) error {
+	cs, err := k8sutil.NewClientset(kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	ns := c.Metadata().Namespace
+	if ns == "" {
+		return fmt.Errorf("component %s namespace is empty", name)
+	}
+
+	cfg, err := HelmActionConfig(ns, kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed preparing helm client: %w", err)
+	}
+
+	var errs []string
+
+	uninstall := action.NewUninstall(cfg)
+
+	resp, err := uninstall.Run(name)
+	if err != nil {
+		errs = append(errs, err.Error())
+	} else {
+		fmt.Println(resp.Info)
+	}
+
+	// Ensure we delete the manually created namespace which was not created by helm
+	if err = cs.CoreV1().Namespaces().Delete(ns, &metav1.DeleteOptions{}); err != nil {
```

Comment:
I wouldn't do that initially. The namespace may contain other resources too.

Reply:
Also, if we remove entire namespace, then there is no point to remove the release itself _I think_. Unless we care about some pre-delete hooks to run.

Reply:
I understand that not deleting the namespace is the original behaviour of the `helm uninstall`. But I was wondering if we might introduce unknown issues due to stale configuration that might be left behind because of undeleted namespace.

Should we add a flag so that if user chooses to do delete the namespace they can delete it? WDYT?

Reply:
> Also, if we remove entire namespace, then there is no point to remove the release itself I think.

I don't understand this? Why we should not bother about deleting a release if we delete the namespace? How do we take care of deleting cluster wide resources?

Reply:
>I don't understand this? Why we should not bother about deleting a release if we delete the namespace? How do we take care of deleting cluster wide resources?

Ah yes, you're right. I forgot that release's object may go beyond single namespace :+1: 

>Should we add a flag so that if user chooses to do delete the namespace they can delete it? WDYT?

I think flag seems reasonable :+1: But it should be off by default.

Reply:
added flag.